Assignment/Assignment/PingProcess.cs

```diff
@@ -27,40 +28,58 @@ public PingResult Run(string hostNameOrAddress)
 
     public Task<PingResult> RunTaskAsync(string hostNameOrAddress)
     {
-        throw new NotImplementedException();
+        Task<PingResult> task = Task.Run(() => Run(hostNameOrAddress));
+
+        return task;
     }
 
     async public Task<PingResult> RunAsync(
         string hostNameOrAddress, CancellationToken cancellationToken = default)
     {
-        Task task = null!;
-        await task;
-        throw new NotImplementedException();
+        while(true)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            Task<PingResult> task = Task.Run(() => RunTaskAsync(hostNameOrAddress), cancellationToken);
+            return await task;
+        }

```

Comment:
You don't necessarily want this to hang forever.

Something like this may be more approperate
```suggestion
        Task<PingResult> task = Task.Run(() =>
        {
            cancellationToken.ThrowIfCancellationRequested();
            Process process = RunProcessInternal( // this will simulate your long running task
                StartInfo, updateStdOutput, default, cancellationToken);
            return new PingResult(process.ExitCode, stringBuilder?.ToString());
        });
        return await task.WaitAsync(cancellationToken);
```